### PR TITLE
Issue #141 Add Vagrantfile to automate dev environment

### PR DESCRIPTION
**Pull request by thirtytwobits**
Problem:
There is no automated way to setup a build environment.

Solution:
Use Vagrant to pull an Ubuntu image and install the same
prerequisites used on Travis.

Testing:
Full build on my macintosh.

#141

### CODE REVIEW

**Comment by thirtytwobits** [comment]
The failure looks like a transient network error. Can you retry the CI run?

**Comment by kjetilkjeka** [comment]
I restarted the failing test

**Comment by pavel-kirienko** [comment]
Good to go?

**Comment by antoinealb** [comment]
Minor nitpick, but bootstrap.sh should probably contain `set -eu` in order to abort if one command fails.

**Comment by thirtytwobits** [comment]
Ya see @antoinealb, it wasn't a nit. You found a real error. Thanks. Fixed.

File: README.md [hunk 1]
```diff
@@ -106,11 +106,37 @@ make
...
+vagrant up
+vagrant ssh
+mkdir build
+cd build
+mkdir build && cd build && cmake .. -DCMAKE_BUILD_TYPE=Debug -DCONTINUOUS_INTEGRATION_BUILD=1
+```
+
+> Note that -DCONTINUOUS_INTEGRATION_BUILD=1 is required for this build as the realtime unit tests will not work on a virt.
+
+You can the build using commands like:
```
Comment (pavel-kirienko) [inline README.md]: You can the what?

File: bootstrap.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+sudo add-apt-repository ppa:team-gcc-arm-embedded/ppa -y
+sudo apt-get update
+sudo apt-get -y install cmake
+sudo apt-get -y install python3
+sudo apt-get -y install git
+sudo apt-get -y install g++-5;
+sudo apt-get -y install gcc-arm-embedded
+
+# Export to tell cmake which native compilers to use.
+export CXX="g++-5" CC="gcc-5";
```
Comment (pavel-kirienko) [inline bootstrap.sh]: Would it not be prudent to choose GCC 7? Its compile-time static analysis capabilities are much improved.

File: libuavcan/include/uavcan/util/templates.hpp [hunk 3]
```diff
@@ -202,7 +202,7 @@ struct EnumMin
 /**
  * Selects larger value
  */
-template <long A, long B>
+template <unsigned long A, unsigned long B>
```
Comment (pavel-kirienko) [inline libuavcan/include/uavcan/util/templates.hpp]: Why unsigned though?

File: README.md (hunk 1 above)
Comment (thirtytwobits) [reply README.md]: erp. Fixed.

**Comment by thirtytwobits** [comment]
I like the change. If everyone else is okay with it then let's merge.

**Pull request merged**